pkg/ocp4/ocp4.go

```diff
@@ -50,18 +60,47 @@ func (ocp4Node *Node) Translate(ocp3Node configv1.NodeConfig) {
 // GenYAML returns the list of translated CRDs
 func (ocp4Master *Master) GenYAML() ([]Manifest, error) {
 	var manifests []Manifest
-	if ocp4Master.OAuth.Kind != "" {
-		manifest := Manifest{Name: "100_CPMA-cluster-config-oauth.yaml", CRD: ocp4Master.OAuth.GenYAML()}
+
+	// Generate yaml for oauth config
+	crd := ocp4Master.OAuth.GenYAML()
+	manifests = appendOauthManifests(ocp4Master.OAuth.Kind, crd, manifests)
+
+	// Generate yaml for oauth secrets
+	manifests = appendOauthSecrets(ocp4Master.Secrets, manifests)
+
+	// Generate yaml for SDN config
+	networkCR := ocp4Master.Network.GenYAML()
+	manifests = appendSDNManifest(networkCR, manifests)
+
+	return manifests, nil
```

Comment:
The refactor is good but the issue I have here is no error is propagated anymore. 
So maybe just remove it, since the append*Manifests log a message if no CRs are generated.